---
int64 len
string[] out
string message
bool success